homeassistant/components/tplink/config_flow.py

```diff
@@ -384,6 +424,85 @@ async def async_step_user_auth_confirm(
             description_placeholders=placeholders,
         )
 
+    async def async_step_camera_auth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that gives the user option to set camera credentials."""
+        errors: dict[str, str] = {}
+        device = self._discovered_device
+        assert device
+        camera_module = device.modules[Module.Camera]
+        if TYPE_CHECKING:
+            # self.host is set by async_step_user and async_step_pick_device
+            assert self.host is not None
+
+        if user_input:
+            live_view = user_input[CONF_LIVE_VIEW]
+            if not live_view:
+                return self._async_create_or_update_entry_from_device(
+                    device, camera_data={CONF_LIVE_VIEW: False}
+                )
+            un = user_input.get(CONF_USERNAME)
+            pw = user_input.get(CONF_PASSWORD)
+            if bool(un) != bool(pw):
+                errors["base"] = "both_or_none"
```

Comment:
Would it be possible to already bail out here? That would allow us to avoid one level of nesting for the happy path.

Reply:
You can't really bail out because the method needs to go on to show the form again on error. I have made other changes to the method to reduce the nesting.